### PR TITLE
[BX-1012][BX-1013] Tooltips

### PR DESCRIPTION
**Pull request by derHowie**
[Figma link (if any):](https://www.figma.com/file/hzfwKOddfRmeVO4KUsG9wM/Rainbow-BX?type=design&node-id=5056-194420&mode=design&t=NFrRn3f8XbUB9bDQ-0)

## What changed (plus any additional context for devs)
Added tooltips to home, send, and swap.

## Screen recordings / screenshots
PoW: https://recordit.co/yuB4wWyn8F

## What to test
Make sure tooltips in the design appear on hover and react appropriately in scrollable areas.

## Final checklist

- [ ] I have tested my changes in a LavaMoat bundle (`yarn build`).
- [ ] I have tested my changes in Chrome & Brave.
- [ ] If your changes are visual, did you check both the light and dark themes?

### CODE REVIEW

File: src/entries/popup/components/AppConnectionMenu/AppConnectionMenu.tsx [hunk 1]
```diff
@@ -215,7 +216,20 @@ export const AppConnectionMenu = ({
...
+          <CursorTooltip
+            align="start"
+            arrowAlignment="left"
+            arrowDirection="up"
+            text={i18n.t('tooltip.switch_network')}
+            textWeight="bold"
+            textSize="12pt"
+            textColor="labelSecondary"
+            marginLeft="2px"
+            marginTop="76px"
```
[the remark this answers is not in the document]
Comment (derHowie) [reply src/entries/popup/components/AppConnectionMenu/AppConnectionMenu.tsx]: This has been addressed, please take another look when you have some time

File: src/entries/popup/pages/send/SendTokenInput.tsx [hunk 2]
```diff
@@ -279,13 +280,34 @@ export const SendTokenInput = React.forwardRef<
...
+              testId={`input-wrapper-close-${'token-input'}`}
+            />
+          </CursorTooltip>
+        ) : (
+          <InputActionButton
+            showClose={!!asset}
+            onClose={onCloseDropdown}
+            onDropdownAction={onDropdownAction}
+            dropdownVisible={dropdownVisible}
+            testId={`input-wrapper-close-${'token-input'}`}
```
Comment (estebanmino) [inline src/entries/popup/pages/send/SendTokenInput.tsx:308]: duplicated code from line 294 ?

File: src/entries/popup/pages/send/ToAddressInput.tsx [hunk 3]
```diff
@@ -429,13 +430,34 @@ export const ToAddressInput = React.forwardRef<InputRefAPI, ToAddressProps>(
...
+                  testId={`input-wrapper-close-to-address-input`}
+                />
+              </CursorTooltip>
+            ) : (
+              <InputActionButton
+                showClose={!!toAddress}
+                onClose={onActionClose}
+                onDropdownAction={onDropdownAction}
+                dropdownVisible={dropdownVisible}
+                testId={`input-wrapper-close-to-address-input`}
```
Comment (estebanmino) [inline src/entries/popup/pages/send/ToAddressInput.tsx:458]: duplicated code from line 444 ?

**Review by estebanmino — COMMENTED** [review]
this needs to be updated to latest tab, we don't have tokens and activity tabs like that anymore
image

**Review by estebanmino — COMMENTED** [review]
maybe i'm crazy but i heard christian asking for the tooltips to show up after 2 or 3 seconds? they're showing right away now

File: src/entries/popup/components/Tooltip/CursorTooltip.tsx [hunk 4]
```diff
@@ -1,55 +1,107 @@
...
-          background="green"
-          style={{
-            position: 'fixed',
-            pointerEvents: 'none',
-            marginTop: marginTop,
-            marginLeft: marginLeft,
+          ref={childElementWrapperRef}
+          onMouseEnter={() => {
+            setIsHovering(true);
+            showTimer.current = setTimeout(() => setOpen(true), 750);
```
Comment (estebanmino) [inline src/entries/popup/components/Tooltip/CursorTooltip.tsx:95]: ah maybe this is the timer?

File: src/entries/popup/components/Tooltip/Tooltip.tsx [hunk 5]
```diff
@@ -85,22 +92,33 @@ export const Tooltip = ({
...
+                  <Inline alignVertical="center" space={'6px'}>
+                    <TextOverflow
+                      color={textColor || 'label'}
+                      size={textSize || '16pt'}
+                      weight={textWeight || 'bold'}
+                    >
+                      {text}
+                    </TextOverflow>
+                    {hint ? <ShortcutHint hint={hint} small /> : null}
+                  </Inline>
```
Comment (estebanmino) [inline src/entries/popup/components/Tooltip/Tooltip.tsx:121]: any reason to not merge Tooltip and CursorTooltip into 1 component? seems like we don't use Tooltip anymore

File: src/entries/popup/components/Tooltip/CursorTooltip.tsx [hunk 6]
```diff
@@ -1,38 +1,53 @@
...
   textSize?: TextStyles['fontSize'];
   textWeight?: TextStyles['fontWeight'];
   textColor?: TextStyles['color'];
+  hint?: string;
 }) => {
   const [open, setOpen] = useState(false);
+
+  if (process.env.IS_TESTING) {
+    return <Box>{children}</Box>;
+  }
```
[the remark this answers is not in the document]
Comment (estebanmino) [reply src/entries/popup/components/Tooltip/CursorTooltip.tsx]: ah seems like this is the case

**Review by estebanmino — APPROVED** [review]
🚀 🚀 🚀 🚀 🚀 🚀

**Pull request merged**